src/arrayMethodJoin.js

```diff
@@ -4,8 +4,37 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
+  [].__proto__.join2 = function(separator = ',') {
     // write code here
+    let sep = separator;
+
+    if (this.length === 0) {
+      return '';
+    }
```

Comment:
there is no need for these lines because in line 27 you have assigned the first element  of the array to the result and  when the array is empty , it will return empty string.